sklearn/neighbors/_distance_metric.py

```diff
@@ -0,0 +1,20 @@
+# TODO: Remove this file in 1.2
+import warnings
+
+from ..metrics import DistanceMetric as _DistanceMetric
+
+
+class DistanceMetric(_DistanceMetric):
+    @classmethod
+    def _warn(cls):
+        warnings.warn(
+            "sklearn.neighbors.DistanceMetric has been moved "
+            "to sklearn.metrics.DistanceMetric in 1.0. "
+            "This import path will be removed in 1.2",
```

Comment:
The policy is to start issuing the deprecation at the next release n and remove the old API in n+2.

```suggestion
            "to sklearn.metrics.DistanceMetric in 1.1. "
            "This import path will be removed in 1.3",
```